ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java

```diff
@@ -254,6 +299,7 @@ private static HttpClient createHttpClient(final Vertx vertx, final ClientOption
         .setSsl(clientOptions.isUseTls())
         .setUseAlpn(clientOptions.isUseAlpn())
         .setProtocolVersion(HttpVersion.HTTP_2)
+        .setHttp2ClearTextUpgrade(false)
```

Comment:
Hey @purplefox I had to set this since otherwise the new tests were failing with
```
java.lang.IllegalStateException: Cannot write an HTTP/2 frame over an HTTP/1.x connection
```
but I don't understand why that is. How come the cleartext upgrade prevents writing of HTTP/2 frames? Is the issue a race condition where the upgrade hasn't taken place by the time I try writing an HTTP/2 frame?

Reply:
Relatedly, do you think this change needs to be backported to the 6.0.x branch? (The old integrations tests pass on 6.0.x without this change so it seems fine to not backport but I wonder if I'm not realizing additional implications of this change.)

Reply:
Tbh I am not an expert on the http2 stuff here. I'd recommend posting on the vert.x google group.